### DISCUSSION TITLE
`npm install` does not create `node_modules` in workspaces

### DISCUSSION THREAD

**Discussion by vicasas** [vercel/turborepo#3397] [Help] [answered]
I created a new repository using the `npx create-turbo@latest` command where I created a new workspace in `packages/apm` which has some dependencies. Following the documentation, it indicates that from the monorepo root you must do `npm install` to install the packages for each workspace, but this is not working.

I would expect that in `apps/web`, `apps/docs` and `packages/apm` running `npm install` from the root of the monorepo would create a `node_modules` for each directory, which is not happening.

But this is working for `packages/ui` that comes by default, I don't know why it works here while in the other workspaces it doesn't.

It is also not creating the `.turbo`.

**Comment by AndyOooh** [comment]
I observed the same. With yarn each workspace gets it's own node_modules but it's not the case with npm. Did you manage to figure out why this is?

**Reply by vicasas** [reply]
@AndyOooh Due to time issues, I really haven't given the topic more thought. I have not tried or carried out tests again either.

**Comment by anthonyshew** [comment, marked as answer]
Depending on your package manager, your package manager configuration, and the dependencies you're installing, you may end up with `node_modules` in your packages or those dependencies may be hoisted to the root. As long as your applications are working as expected for your local development and deployments, your package manager is doing it's job correctly (managing packages!).

I just tried `create-turbo` with npm and everything appears to work as expected for me. I'm going to opt to close this discussion. If you think you've found a Turbo-based bug, please make us an issue with a reproduction so we can investigate!
